tests/unit/pushnotifications/test_pushservice.py

```diff
@@ -0,0 +1,167 @@
+import json
+import time
+
+import firebase_admin
+from firebase_admin import messaging
+
+from relay.blockchain.currency_network_events import (
+    TransferEvent,
+    TrustlineRequestEvent,
+    TrustlineUpdateEvent,
+)
+from relay.events import MessageEvent
+from relay.pushservice.pushservice import (
+    FirebaseRawPushService,
+    _build_data_message,
+    _build_data_prop,
+)
+
+from .utils import MockAdapter, MockCredential
+
+requestMessagePayload = (
+    '{"type":"PaymentRequest","networkAddress":"0x12657128d7fa4291647eC3b0147E5fA6EebD388A",'
+    '"from":"0xB6cD40e87d1ED3eCd455cDa0B6EA9FD56F83f0a7",'
+    '"to":"0xE85bd548b2C961A2385628dcbBcc9A2E480dD925","amount":{"decimals":8,"raw":"1000000000",'
+    '"value":"10"},"subject":"test subject","id":"0x5d57fabcc8b6055b",'
+    '"counterParty":"0xB6cD40e87d1ED3eCd455cDa0B6EA9FD56F83f0a7","direction":"received",'
+    '"user":"0xE85bd548b2C961A2385628dcbBcc9A2E480dD925"} '
+)
+
+message_event = MessageEvent(
+    message=requestMessagePayload, type="PaymentRequest", timestamp=int(time.time())
+)
+
+cred = MockCredential()
+all_options = {"projectId": "explicit-project-id"}
+admin_app = firebase_admin.initialize_app(cred, all_options)
+
+_DEFAULT_RESPONSE = json.dumps({"name": "message-id"})
+
+
+def _instrument_messaging_service(app, status=200, payload=_DEFAULT_RESPONSE):
+    fcm_service = messaging._get_messaging_service(app)
+    recorder = []
+    fcm_service._client.session.mount(
+        "https://fcm.googleapis.com", MockAdapter(payload, status, recorder)
+    )
+    return fcm_service, recorder
+
+
+def test_send_on_blockchain_event(web3_event_transfer):
+    raw_service = FirebaseRawPushService(app=admin_app)
```

Comment:
I would have put them (raw_service and recorder) also in a fixture, but it is a minor detail.